test/models/ocelot/stanzas/ChoiceStanzaSpec.scala

```diff
@@ -187,11 +187,189 @@ class ChoiceStanzaSpec extends BaseSpec {
 
   }
 
+  "Choice" must {
+    "be creatable from a ChoiceStanza " in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("4", LessThanOrEquals, "3"), ChoiceStanzaTest("3", LessThanOrEquals, "4")), false)
+      val choice = Choice(stanza)
+      choice.next shouldBe stanza.next
+      choice.tests.zipWithIndex.foreach{
+        case (LessThanOrEqualsTest(_,_), index) if stanza.tests(index).test == LessThanOrEquals => succeed
+        case x => fail
+
+      }
+    }
+
+    "Evaluate to correct result when one of the tests succeed" in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("4", LessThanOrEquals, "3"),
+                                                        ChoiceStanzaTest("3", LessThanOrEquals, "4")), false)
+      val choice = Choice(stanza)
+      val lc = LabelCache()
+      val expectedResult = (Seq("41"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+    "Evaluate to correct result when no tests succeed" in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("4", LessThanOrEquals, "3"),
+                                                        ChoiceStanzaTest("3", MoreThan, "4")), false)
+      val choice = Choice(stanza)
+      val lc = LabelCache()
+      val expectedResult = (Seq("50"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+    "Evaluate to correct result when one of the tests succeed referencing labels" in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("[label:X]", LessThanOrEquals, "[label:Y]"),
+                                                        ChoiceStanzaTest("3", LessThanOrEquals, "4"),
+                                                        ChoiceStanzaTest("3", NotEquals, "4")), false)
+      val choice = Choice(stanza)
+      val labels = Map("X"->Label("X", Some("33.5")), "Y"->Label("Y", Some("44")))
+      val lc = LabelCache(labels)
+      val expectedResult = (Seq("40"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+    "Evaluate to correct result when no tests succeed referencing labels" in {
```

Comment:
This produces an interesting result. With three tests I think next should have four entries. The last one being used if all tests fail. I'm not certain what happens if there are three tests and three destinations and all of the tests fail.

Reply:
Ah yes , test succeeds because the third of the ChoiceStanzaTests succeeds, if it had failed, it would have blown up because there arent 4 values in the next, will fix